Assets/Xbox Live/Scripts/Stats/StatBase.cs

```diff
@@ -34,6 +38,37 @@ private void Awake()
         // Ensure that a StatsManager has been created so that stats will be sync with the service as they are modified.
         var statsManager = StatsManagerComponent.Instance;
     }
+
+    void Start()
+    {
+        if (XboxLiveUserHelper.Instance.SingleUserModeEnabled)
+        {
+            this.XboxLiveUser = XboxLiveUserHelper.Instance.SingleXboxLiveUser;
+            if (this.XboxLiveUser == null || this.XboxLiveUser.User == null || !this.XboxLiveUser.User.IsSignedIn)
+            {
+                StatsManagerComponent.Instance.LocalUserAdded += (sender, args) =>
+                {
+                    this.HandleGetStat(args.User, this.Name);
+                };
+            }
+        }
+    }
+
+    protected void Update()
+    {
+        if (this.XboxLiveUser != null && this.XboxLiveUser.User != null && this.XboxLiveUser.User.IsSignedIn && !this.isLocalUserAdded)
+        {
+            StatsManagerComponent.Instance.LocalUserAdded += (sender, args) =>
+            {
+                if (args.User.Gamertag == this.XboxLiveUser.User.Gamertag)
+                {
+                    this.HandleGetStat(args.User, this.Name);
+                }
+            };
```

Comment:
The problem is at times during start(), the XboxLiveUser is still null (in the case of multi-users, the User is set when sign in happens ... till then it's null) and so the check: if (args.User.Gamertag == this.XboxLiveUser.User.Gamertag) will fail with a null reference exception. I'm adding a flag though to make sure this is only done once.

---
In reply to: [119976341](https://github.com/Microsoft/xbox-live-unity-plugin/pull/66#discussion_r119976341) [](ancestors = 119976341)